src/toolchain/export-web.md

```diff
@@ -0,0 +1,155 @@
+The reason that a nightly `rustc` is required is that nightly is required to build `std` (`-Zbuild-std`).
+Assuming that Rust was installed with `rustup`, this is quite simple.
+
+  ```sh
+  rustup toolchain install nightly
+  rustup component add rust-src --toolchain nightly
+  rustup target add wasm32-unknown-emscripten --toolchain nightly
+  ```
+
+Next, install Emscripten.  The simplest way to achieve this is to install `emsdk` from the git repo. Recommended to use version 3.1.39 for now[^1]
+
+```sh
+git clone https://github.com/emscripten-core/emsdk.git
+cd emsdk
+./emsdk install 3.1.39
+./emsdk activate 3.1.39
+source ./emsdk.sh     (or ./emsdk.bat on windows)
+```
+
+It would also be HIGHLY recommended to follow the instructions in the terminal to add `emcc`[^2] to your `PATH`.
+If not, it is necessary to manually `source` the `emsdk.sh` file in every new terminal prior to compilation.
+This is platform-specific.
+
+---
+
+
+## Project Configuration
+
+Enable the `experimental-wasm` feature on `gdext` in the `Cargo.toml` file.
+It is also strongly recommended to enable the `lazy-function-tables` feature to avoid an extremely long compile time with release builds.
+It is likely to due a bug, but is still being investigated.  Edit the line to something like the following:
+
+```toml
+godot = { git = "https://github.com/godot-rust/gdext", branch = "master", features = ["experimental-wasm", "lazy-function-tables"] }
+```
+
+If you do not already have `.cargo/config.toml` file, do the following:
+
+- Create a `.cargo` directory at the same level as your `Cargo.toml`.
+- Inside of that directory, create a `config.toml` file.
+
+The contents of this file need to contain the following:
+
+```toml
+[target.wasm32-unknown-emscripten]
+rustflags = [
+    "-C", "link-args=-sSIDE_MODULE=2",
+    "-C", "link-args=-sUSE_PTHREADS=1",
+    "-C", "target-feature=+atomics,+bulk-memory,+mutable-globals",
+    "-Zlink-native-libraries=no"
+]
+```
+
+Edit the project's `.gdextension` file to include support for web exports.
+This file will probably be at `godot/{YourCrate}.gdextension`.
+The format will be similar to the following:
+
+```gdextension
+web.debug.wasm32 = "res://../rust/target/wasm32-unknown-emscripten/debug/EXTENSIONNAME.wasm"
+web.release.wasm32 = "res://../rust/target/wasm32-unknown-emscripten/release/EXTENSIONNAME.wasm"
+```
+
+---
+
+
+## Compile the Project
+
+Verify `emcc` is in the `PATH`. This can be as simple as doing the following:
+
+```sh
+emcc --version
+```
+
+Compile the code.
+It is necessary to both use the nightly compiler and specify to build std[^3], along with specifying the Emscripten target.
+
+```sh
+cargo +nightly build -Zbuild-std --target wasm32-unknown-emscripten
+```
+
+---
+
+
+## Godot Editor Setup
+
+Add a web export in the Godot Editor. In the top menu bar, go to `Project > Export...` and configure it there.
+Make sure to turn on the `Extensions Support` checkbox.
```

Comment:
Should we mention to download export templates if they are not yet configured?

Reply:
I think you get a prompt if it's missing to download it, and should be downloaded at the same time you'd download templates for any other export.  I'd have to intentionally trigger that and see about that.

Reply:
From the last time I tried it in my VM, I believe it displays some red text at the bottom suggesting you need Export Templates, but I think you still need to click some buttons to prompt the download. Could be useful to list the steps needed, but very briefly.

Reply:
I see. It's fine to not explicitly mention it then.